packages/application-generic/src/services/analytics.service.ts

```diff
@@ -102,11 +104,22 @@ export class AnalyticsService {
         data,
       });
 
-      this.segment.track({
-        userId: userId,
-        event: name,
-        properties: data,
-      });
+      try {
+        this.segment.track({
+          userId: userId,
+          event: name,
+          properties: data,
+        });
+      } catch (error: any) {
+        Logger.error(
+          JSON.stringify({
+            name: name,
+            usedId: userId,
+            message: error.message,
```

Comment:
I am wondering if we could add more data here, like the transaction id etc..
one way of doing it is passing it as part of the data to track function. and then add it here 
`transactionId: data. transactionId,`

Reply:
I think it is a good idea. Also we then so can start to design the interface of the `data` passed. 👍🏻

Reply:
I think that pino supporting objects, so we don't need to pass stringify there. We can have an error name, and pass the whole object.

In most cases where transacitonId is available, we automatically add it to the logger context. So we don't need to manually specify it here. It will be automatically added :)